File: src/02_datatypes.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import numpy as np
import datetime

# ----------
# Data Types
# ----------
bvar = True
nvar = 42.0
ivar = 42
svar = "Hello World"
dtvar = datetime.datetime(2017, 2, 28, 12, 0, 0)
dvar = datetime.date(2017, 2, 28)
tvar = datetime.time(12, 0, 0)

print("Boolean:", bvar, type(bvar))
print("Floating Point:", nvar, type(nvar))
print("Integer:", ivar, type(ivar))
print("String:", svar, type(svar))
print("Datetime:", dtvar, type(dtvar))
print("Date:", dvar, type(dvar))
print("Time:", tvar, type(tvar))

# ------
# Arrays
# ------
x = [1, 2, 3, 4]
# Try to use multiplication on the list.
print(x*2)
# Use numpy for multiplication.
print(np.asarray(x)*2)

# List comprehension: Double the value for each entry in the list.
print([a*2 for a in x])
# List comprehension: Double the value for each entry in the list where the value <= 2.
print([a*2 for a in x if a<=2])

File: src/02_datatypes.r
library(data.table)
library(psych)

setwd("~/sas-py-r/src")

# ----------
# Data Types
# ----------

bvar = TRUE
nvar = 42.0
ivar = as.integer(42)
svar = "Hello World"
dtvar = as.POSIXct("2017-02-28 12:00:00 PM", format = "%Y-%m-%d %I:%M:%S %p")
dvar = as.Date("2017-02-28")

print(paste("Boolean:", bvar, class(bvar)))
print(paste("Floating Point:", nvar, class(nvar)))
print(paste("Integer:", ivar, class(ivar)))
print(paste("String:", svar, class(svar)))
print(paste("Datetime:", dtvar, class(dtvar)))
print(paste("Date:", dvar, class(dvar)))

# ------
# Arrays
# ------
x <- c(1, 2, 3, 4)
print(x*2) # Output is 2 4 6 8